gdscript/sidebar.gd

```diff
@@ -333,55 +233,57 @@ func init(end_task: bool = true) -> void:
 	history_tree.allow_rmb_select = true
 	inspector.node_hovered.connect(_on_node_hovered)
 	inspector.node_unhovered.connect(_on_node_unhovered)
+	%CopyProjectIDButton.pressed.connect(_on_copy_project_id_button_pressed)
 
-	if not check_and_prompt_for_user_name(self._check_for_user_branch):
-		return
-
-	if not TURN_ON_USER_BRANCH_PROMPT:
-		return
-	var timeout = 5.0
-	var timer = Timer.new()
-	timer.wait_time = timeout
-	timer.one_shot = true
-	timer.timeout.connect(self._check_for_user_branch)
-	add_child(timer)
-	timer.start()
-
-func _on_sync_status_icon_pressed():
-	var sync_info = GodotProject.get_sync_server_connection_info()
-	var checked_out_branch = GodotProject.get_checked_out_branch()
+func _try_init():
+	var godot_project = Engine.get_singleton("GodotProject")
+	if godot_project:
+		if !godot_project.has_project():
+			print("Not initialized, showing init panel")
+			print("Patchwork: Updating UI due to init...")
+			update_ui()
+			return
+		else:
+			print("Initialized, hiding init panel")
+			wait_for_checked_out_branch()
+	else:
+		print("No GodotProject singleton!!!!!!!!")
 
-	print("Sync info ===========================", )
-	print("is connected: ", sync_info.is_connected)
-	print("last received: ", human_readable_timestamp(sync_info.last_received * 1000.0))
-	print("last sent: ", human_readable_timestamp(sync_info.last_sent * 1000.0))
+func _process(delta: float) -> void:
+	if deferred_highlight_update:
+		var c = deferred_highlight_update
+		deferred_highlight_update = null
+		c.call()
 
+	if waiting_callables.size() > 0:
+		var callables = waiting_callables.duplicate()
+		for callable in callables:
+			callable.call()
+		waiting_callables.clear()
 
-	if checked_out_branch && sync_info.doc_sync_states.has(checked_out_branch.id):
-		var doc_sync_state = sync_info.doc_sync_states[checked_out_branch.id]
+func init() -> void:
+	print("Sidebar initialized!")
+	print("Patchwork: Updating UI due to init...")
+	update_ui()
 
-		print(checked_out_branch.name, ":")
-		print("  acked heads: ", doc_sync_state.last_acked_heads)
-		print("  sent heads: ", doc_sync_state.last_sent_heads)
-		if doc_sync_state.last_sent != null:
-			print("  last sent: ", human_readable_timestamp(doc_sync_state.last_sent * 1000.0))
-		else:
-			print("  last sent: -")
-		if doc_sync_state.last_received != null:
-			print("  last received: ", human_readable_timestamp(doc_sync_state.last_received * 1000.0))
-		else:
-			print("  last received: -")
+	# Here, the user could easily just hit X and remain anonymous. This can only happen in the case
+	# of a project loaded from a file, where the user's config hasn't been set.
+	# If we want to force the user to enter a username, we could do `while(!require_user_name()): pass`.
+	# But that seems bad.
+	require_user_name()
```

Comment:
While you can't stop a dialog from being closed, you can attach a listener to the `close_requested` signal on the dialog box to automatically re-open the dialog